util/settings/settings.go

```diff
@@ -1093,6 +1105,7 @@ func updateSettingsFromConfigMap(settings *ArgoCDSettings, argoCDCM *apiv1.Confi
 	if settings.PasswordPattern == "" {
 		settings.PasswordPattern = common.PasswordPatten
 	}
+	settings.InClusterServerAddressAllowed = argoCDCM.Data[inClusterServerAddressAllowedKey] == "true"
```

Comment:
I think this is a bug. If this setting doesn't exist, it should return true. So this should probably be:

```
settings.InClusterServerAddressAllowed = argoCDCM.Data[clusterInClusterEnabled] != "false"
```